distributed/tests/test_worker.py

```diff
@@ -425,6 +425,25 @@ async def test_io_loop(cleanup):
             assert w.io_loop is s.loop
 
 
+@gen_cluster(client=True)
+async def test_memory_leak(c, s, a, b):
+    import numpy as np
+    import time
+
+    x = {}
+    def memory_leaking_fn(data):
+        # leak 12MB blocks
+        x[data] = np.random.randint(100, size=12 * 1024 ** 2 // 8)
+
+        time.sleep(0.1)
+        return data
+
+    futures = c.map(memory_leaking_fn, range(1000))
+
+    for f in futures:
+        print(f.result())
```

Comment:
We probably can't `print` in tests, these need to be fully automated.  However in this case I think that just awaiting all of them will probably be enough.  

I'm curious how long this test takes.  Most tests finish in less than a second.  It is important to keep the test suite snappy so that people continue to use it in their daily development cycle.

I suspect that for this to work we need to use `Nanny`s, and be careful about different memory spaces (see `gen_cluster(Worker=Nanny)` and be careful about where `x` lives exactly.  We'll need to make sure that the leaking memory is in the worker process, not in the main client process.

Reply:
As this requires a very exactly defined localcluster, is it fine to generate the cluster inside the test here?
```python
cluster =dask.distributed.LocalCluster(n_workers=2, threads_per_worker=1, memory_limit="512M")
client = dask.distributed.Client(cluster)
```
Also, are 512MB memory fine in the tests, in consideration of the cloud resources?

Another problem is that this test can result in a deadlock.
We should consider this test as failed if it takes longer than <fill_value_here>.
Is this possible somehow?

Reply:
Yes, this test can take a couple of seconds.
I saw that there was a decorator to sign long-running tests.
I should mask this test accordingly, right?

Reply:
> is it fine to generate the cluster inside the test here?

I would encourage you to continue to use `gen_cluster` rather than LocalCluster.  It does many things to ensure that the test cleans up nicely.  The `gen_cluster` decorator has some useful keywords that might help here:

```python
@gen_cluster(Worker=Nanny, nthreads=[("127.0.0.1", 1), ("127.0.0.1", 1)], worker_kwargs={"memory_limit": "512MB"})
```

Yes, marking a slow test with `@pytest.mark.slow` is good if we can't make the test fast

Reply:
Should be resolved now I guess